### PR TITLE
[AMDGPU] Select gfx1150 SALU Float instructions

### PR DESCRIPTION
**Pull request by mbrkusanin**

### CODE REVIEW

**Comment by mbrkusanin** [comment]
Note that this contains https://github.com/llvm/llvm-project/pull/66884
Not sure how to do stacked PRs from fork.

**Comment by jayfoad** [comment]
Why isn't @llvm/pr-subscribers-amdgpu being added automatically as a reviewer?

File: llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp [hunk 1]
```diff
@@ -1211,36 +1211,104 @@ int AMDGPUInstructionSelector::getS_CMPOpcode(CmpInst::Predicate P,
     }
   }
 
-  if (Size != 32)
-    return -1;
+  if (Size == 32) {
```
Comment (rampitec) [inline llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp:1214]: This 'if' is not needed, you have just exited if Size != 32.

**Review by rampitec — APPROVED** [review]

**Comment by mbrkusanin** [comment]
Rebase

File: llvm/test/CodeGen/AMDGPU/code-size-estimate.ll [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ; RUN: llc -march=amdgcn -mcpu=gfx900 -show-mc-encoding < %s | FileCheck -check-prefixes=CHECK,GFX9 %s
 ; RUN: llc -march=amdgcn -mcpu=gfx1030 -show-mc-encoding < %s | FileCheck -check-prefixes=CHECK,GFX10 %s
 ; RUN: llc -march=amdgcn -mcpu=gfx1100 -show-mc-encoding < %s | FileCheck -check-prefixes=CHECK,GFX11 %s
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/code-size-estimate.ll:3]: You missed adding the GFX1100 prefix to this RUN line

Comment (mbrkusanin) [reply llvm/test/CodeGen/AMDGPU/code-size-estimate.ll:3]: https://github.com/llvm/llvm-project/commit/a657deb42ebdfb5e72d0823001485a97ee65a2c6

Comment (jayfoad) [reply llvm/test/CodeGen/AMDGPU/code-size-estimate.ll:3]: Thanks!

**Pull request merged**